irc/utils/net.go

```diff
@@ -6,13 +6,16 @@ package utils
 
 import (
 	"net"
+	"regexp"
 	"strings"
 )
 
 var (
 	// subnet mask for an ipv6 /128:
 	mask128             = net.CIDRMask(128, 128)
 	IPv4LoopbackAddress = net.ParseIP("127.0.0.1").To16()
+
+	validHostnameLabelRegexp = regexp.MustCompile(`^[0-9A-Za-z.\-]+$`)
```

Comment:
should the `.` in the regex here be escaped? (e.g. be changed to `^[0-9A-Za-z\.\-]+$` ?). seems like it might match any character right now

Reply:
`.` in a character class is a literal period: https://stackoverflow.com/a/19976308

This is checked by one of the new tests (the one with the percent sign in the label)

Reply:
ah perfect, awesome then!